astropy/coordinates/sky_coordinate.py

```diff
@@ -362,7 +363,41 @@ def _parse_inputs(self, args, kwargs):
 
         return valid_kwargs
 
-    def transform_to(self, frame):
+    def transforms(self, other, accepting_defaults=False):
```

Comment:
> but there should be some way to transform an other object into self's frame.

The idea is that you would use the following (with a new idea for the kwarg name):
```
other_in_self_frame = other.transform_to(self, attribute_priority='destination')
```
So this says to use the destination attributes if they exist at all, otherwise fall back to source.  Does this work for the use cases here?  This formalism is a bit more symmetric because you would also have the option of using `attribute_priority='source'`.